types/node/module.d.ts

```diff
@@ -107,6 +108,96 @@ declare module 'module' {
              */
             resolve?(specified: string, parent?: string | URL): Promise<string>;
         }
+        namespace NodeJSLoaders {
+            type ModuleFormat = 'builtin' | 'commonjs' | 'json' | 'module' | 'wasm';
+            type ModuleSource = string | ArrayBuffer | NodeJS.TypedArray;
+            interface GlobalPreloadContext {
+                port: MessagePort;
+            }
+            type GlobalPreloadHook = (context: GlobalPreloadContext) => string;
+            interface ResolveHookContext {
+                /**
+                 * Export conditions of the relevant `package.json`
+                 */
+                conditions: string[];
+                /**
+                 *  An object whose key-value pairs represent the assertions for the module to import
+                 */
+                importAssertions: Object;
+                /**
+                 * The module importing this one, or undefined if this is the Node.js entry point
+                 */
+                parentURL: string | undefined;
```

Comment:
```suggestion
                parentURL?: string | undefined;
```
(unless there was a reason)

Reply:
As I see, `context` has field `parentURL`, it can be `undefined` but not optional
https://github.com/nodejs/node/blob/802c52fb8b3c5853d683adf6612041b2e5eed6dd/lib/internal/modules/esm/hooks.js#L247C7-L247C16